MapboxCoreNavigation/NavigationLocationManager.swift

```diff
@@ -17,24 +17,48 @@ open class NavigationLocationManager: CLLocationManager {
     
     var lastKnownLocation: CLLocation?
     
+    /**
+     Indicates whether the device is plugged in or not.
+     */
+    public private(set) var isPluggedIn: Bool = false
+    
+    /**
+     Indicates whether the location manager’s desired accuracy should update
+     when the battery state changes.
+     */
+    public var automaticallyUpdatesDesiredAccuracy = true
+    
     override public init() {
         super.init()
         
-        let always = Bundle.main.locationAlwaysUsageDescription
-        let both = Bundle.main.locationAlwaysAndWhenInUseUsageDescription
-        
-        if always != nil || both != nil {
-            requestAlwaysAuthorization()
-        } else {
-            requestWhenInUseAuthorization()
-        }
+        requestWhenInUseAuthorization()
         
         if #available(iOS 9.0, *) {
             if Bundle.main.backgroundModes.contains("location") {
                 allowsBackgroundLocationUpdates = true
             }
         }
         
-        desiredAccuracy = kCLLocationAccuracyBestForNavigation
+        desiredAccuracy = kCLLocationAccuracyBest
+        
+        #if os(iOS)
+            UIDevice.current.addObserver(self, forKeyPath: "batteryState", options: [.initial, .new], context: nil)
```

Comment:
The block-based KVO is the recommended approach as of Swift 3.2. It's arguably cleaner, contained in one place, and implicitly removed.

https://github.com/mapbox/mapbox-navigation-ios/blob/1356176ea13c17dc6564d8270a10e34808a2019e/MapboxNavigation/NavigationMapView.swift#L228-L230